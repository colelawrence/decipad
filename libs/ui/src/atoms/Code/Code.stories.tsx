import { Meta, Story } from '@storybook/react';
import { Code } from './Code';

export default {
  title: 'Atoms / Editor / Text / Inline / Code',
  component: Code,
  args: {
    children: 'Text',
  },
} as Meta;

export const Normal: Story<{ children: string }> = (args) => <Code {...args} />;
