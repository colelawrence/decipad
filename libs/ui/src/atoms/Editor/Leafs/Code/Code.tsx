import { css } from '@emotion/react';
import { PlatePluginComponent } from '@udecode/plate';
import { code, cssVar } from '../../../../primitives';

const codeStyles = css(code, {
  borderRadius: '100vh',
  backgroundColor: cssVar('highlightColor'),
  padding: '6px 12px',
  margin: '0 6px',
});

export const CodeLeaf: PlatePluginComponent = ({ attributes, children }) => {
  return (
    <code css={codeStyles} {...attributes}>
      {children}
    </code>
  );
};
