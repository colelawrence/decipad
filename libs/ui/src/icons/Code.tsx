import type { SVGProps } from 'react';

export const Code = (props: SVGProps<SVGSVGElement>) => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      fill="none"
      viewBox="0 0 16 16"
      role="img"
      {...props}
    >
      <title>Code</title>
      <path
        fill="currentColor"
        fillOpacity={0.1}
        stroke="currentColor"
        strokeWidth={1.2}
        d="M2 12V4a2 2 0 0 1 2-2h8a2 2 0 0 1 2 2v8a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2Z"
      />
      <path
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        d="M5.563 7.063 7.438 8.75l-1.875 1.688"
      />
    </svg>
  );
};
