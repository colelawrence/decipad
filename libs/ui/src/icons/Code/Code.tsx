import { FC } from 'react';
import { cssVar } from '../../primitives';

export const Code = (): ReturnType<FC> => {
  return (
    <svg viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
      <title>Code</title>
      <path
        d="M11.9375 2.5625H4.0625C3.23407 2.5625 2.5625 3.23407 2.5625 4.0625V11.9375C2.5625 12.7659 3.23407 13.4375 4.0625 13.4375H11.9375C12.7659 13.4375 13.4375 12.7659 13.4375 11.9375V4.0625C13.4375 3.23407 12.7659 2.5625 11.9375 2.5625Z"
        fill={cssVar('iconBackgroundColor')}
        stroke={cssVar('currentTextColor')}
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M5.5625 7.0625L7.4375 8.75L5.5625 10.4375"
        stroke={cssVar('currentTextColor')}
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
};
