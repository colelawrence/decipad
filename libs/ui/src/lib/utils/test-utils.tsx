import { ChakraProvider } from '@chakra-ui/react';
import { render, RenderOptions } from '@testing-library/react';
import React from 'react';
import { theme } from '../theme';

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <ChakraProvider resetCSS theme={theme}>
      {children}
    </ChakraProvider>
  );
};

const customRender = (
  ui: JSX.Element,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
