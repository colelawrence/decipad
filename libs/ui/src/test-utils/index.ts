export * from './menu';
export * from './language';
