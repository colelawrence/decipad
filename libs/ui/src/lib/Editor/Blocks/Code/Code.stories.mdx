import { Meta, Story, Canvas, Props } from '@storybook/addon-docs/blocks';
import { Code } from './Code.component';

<Meta
  title="Legacy/Editor/Blocks/Code"
  parameters={{
    docs: {
      source: {
        code: `<Code
  element={{
    children: [{ text: 'const hello = "Hello World";' }],
  }}
  attributes={{ 'data-slate-node': 'element', ref: null }}
>
  const hello = "Hello World";
</Code>`,
      },
    },
  }}
  argTypes={{
    children: {
      defaultValue: 'const hello = "Hello World";',
      description: 'The text inside of the Inline code component',
      type: 'string',
    },
  }}
  component={Code}
/>

# Code

The Code component serves as one of the editor's slate blocks which is basically just an inline code component to make the text of that component look like some sort of code.

export const Template = (args) => <Code {...args} />;

<Canvas>
  <Story name="Code">{Template.bind({})}</Story>
</Canvas>

<Props story="Code" />
