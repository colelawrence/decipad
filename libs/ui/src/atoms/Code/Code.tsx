import { css } from '@emotion/react';
import { ReactNode } from 'react';
import { code, cssVar, p16Regular } from '../../primitives';
import { SlateLeafProps } from '../../utils';

const styles = css(code, {
  backgroundColor: cssVar('highlightColor'),
  borderRadius: '100vmax',
  padding: `calc((${p16Regular.fontSize} - ${code.fontSize}) / 2)`,
  margin: '0 6px',
});

interface CodeProps extends SlateLeafProps {
  readonly children: ReactNode;
}
export const Code = ({ children }: CodeProps): ReturnType<React.FC> => {
  return <code css={styles}>{children}</code>;
};
