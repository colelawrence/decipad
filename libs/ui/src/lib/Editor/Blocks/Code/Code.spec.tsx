import React from 'react';
import { render } from 'test-utils';
import { Code } from './Code.component';

describe('Code Block', () => {
  it('renders', () => {
    const { container } = render(
      <Code
        element={{
          children: [{ text: 'const hello = "Hello World";' }],
        }}
        attributes={{ 'data-slate-node': 'element', ref: null }}
      >
        const hello = "Hello World";
      </Code>
    );

    expect(container).toBeDefined();
  });

  it('matches snapshot', () => {
    const { container } = render(
      <Code
        element={{
          children: [{ text: 'const hello = "Hello World";' }],
        }}
        attributes={{ 'data-slate-node': 'element', ref: null }}
      >
        const hello = "Hello World";
      </Code>
    );

    expect(container).toMatchSnapshot();
  });
});
