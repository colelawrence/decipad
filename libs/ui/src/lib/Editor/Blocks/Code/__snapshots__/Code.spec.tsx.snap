// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Block matches snapshot 1`] = `
.emotion-0 {
  display: inline-block;
  font-family: var(--chakra-fonts-mono);
  font-size: var(--chakra-fontSizes-sm);
  -webkit-padding-start: 0.2em;
  padding-inline-start: 0.2em;
  -webkit-padding-end: 0.2em;
  padding-inline-end: 0.2em;
  border-radius: var(--chakra-radii-sm);
  background: var(--chakra-colors-gray-100);
  color: var(--chakra-colors-gray-800);
}

<div>
  <code
    class="chakra-code emotion-0"
    data-slate-node="element"
  >
    const hello = "Hello World";
  </code>
</div>
`;
